README.md

```diff
@@ -0,0 +1,144 @@
+# Enigma
+
+Enigma is a Ruby gem that fetches your application's passwords from AWS Secrets Manager and builds a file containing them using a `Templatefile` as the guide. This results in a simple and platform-agnostic approach to rendering your application password in to a compilable file.
+
+## Installation
+
+`gem install enigma-gen`
+
+## Usage
+
+Once the gem is installed, simply running `enigma exec` is enough to get you started. However, should you want to pass in specific command line options instead of specifying values in your `Enigmafile`, you can do that as well.
+
+```
+--access_key_id [String] Your AWS Access Key ID
+```
+```
+--secret_access_key [String] Your AWS Secret Access Key
+```
+```
+--aws_region [String] Your AWS Region
+```
+```
+--environment [String] Your specified secret environment
+```
+```
+--config [String] Path to your Enigmafile
+```
+
+**Note**: Parameters passed in via CLI take precedence over those listed in your `Enigmafile`.
+
+## Enigmafile
+
+The Enigmafile is used to tell Enigma what to do. It is a YAML file with a basic structure and contains the following properties:
+
+1. **access_key_id**  
+  The environment variable that your `AWS_ACCESS_KEY_ID` is associated with. Do **not** commit your actual access key ID to the Enigmafile
+2. **secret_access_key**  
+  The environment variable that your `AWS_SECRET_ACCESS_KEY` is associated with. Do **not** commit your actual secret access key to the Enigmafile
+3. **aws_region**  
+  The AWS region that your Secrets Manager instance is in. Eg: `us-east-2`
+4. **template_path**  
+  The relative path from the project root to the file containing your template
+5. **output_path**  
+  The relative path from the project root to the desired location of the file containing your application's passwords
+6. **environment**  
+  Optional. If you've given your passwords an environment in Secrets Manager, add the environment here. Eg: `prod/someProvider`, enter `prod`. This will prefix **all** searches in Secrets Manager with the environment value and a trailing slash.
+7. **allow_missing_items**  
+  Optional. Will allow Enigma to continue if a secret listed in your Enigmafile is not found in Secrets Manager
```

Comment:
Are there default values for these?

Reply:
6. No
7. Yes, false. I've updated the docs.